src/networkcloud/azext_networkcloud/operations/common_ssh.py

```diff
@@ -0,0 +1,144 @@
+                    raise InvalidArgumentValueError(
+                        f"No public keys found in the path: {path}"
+                    )
+            elif os.path.isfile(path):
+                with open(path, "r", encoding="utf-8") as k:
+                    content = k.read()
+                try:
+                    if keys.is_valid_ssh_rsa_public_key(content):
+                        key_list.append({"keyData": content})
+                except Exception as excep:
+                    raise InvalidArgumentValueError(
+                        f"Unsupported Key {path} is provided.\nContent:\n{content}"
+                    ) from excep
+            else:
+                raise InvalidArgumentValueError(
+                    f"An invalid directory or key was provided: {path}"
+                )
+        return key_list
+
+    @staticmethod
+    def add_ssh_key_action(values):
+        """validate and add ssh key to the list"""
+
+        key_data = []
+        for key in values:
+            # Casting is necessary to get away from the custom type AAZSimpleType wrapping the string we want
+            key = str(key)
+            try:
+                if keys.is_valid_ssh_rsa_public_key(key):
+                    key_data.append({"keyData": key})
+            except Exception as excep:
+                raise InvalidArgumentValueError(
+                    f"Unsupported Key is provided.\nContent:\n{key}"
+                ) from excep
+        return key_data
+
+    @staticmethod
+    def build_ssh_arg_schema(args_schema, group):
+        args_schema.ssh_key_values = AAZListArg(
+            options=["--ssh-key-values"],
+            arg_group=group,
+            help="The list of space-separated SSH public keys.",
+        )
+        args_schema.ssh_key_values.Element = AAZStrArg()
+        args_schema.generate_ssh_keys = AAZBoolArg(
+            options=["--generate-ssh-keys"],
+            arg_group=group,
+            help="Generate SSH public and private key files if missing. The keys will "
+            + "be stored under '~/.ssh'.",
+        )
+        args_schema.generate_ssh_keys.Element = AAZStrArg()
+        args_schema.ssh_dest_key_path = AAZListArg(
+            options=["--ssh-dest-key-path"],
+            arg_group=group,
+            help="The list of space-separated paths to ssh public keys. A file path can be "
+            + "provided or a directory. If a directory is provided will attempt to load all the "
+            + "public keys (.pub) found.",
+        )
+        args_schema.ssh_dest_key_path.Element = AAZStrArg()
+        return args_schema
+
+    @staticmethod
+    def add_ssh_config(args):
+        ssh_keys = []
+        if bool(args.generate_ssh_keys):
+            ssh_keys += CustomSshOptions.generate_ssh_keys()
+        if list(args.ssh_dest_key_path):
+            ssh_keys += CustomSshOptions.get_ssh_keys_from_path(
+                list(args.ssh_dest_key_path)
+            )
```

Comment:
You can use to_serialized_data() function, which will serialize an AAZValue to python build-in type value.
```suggestion
        if has_value(args.ssh_dest_key_path) and args.ssh_dest_key_path.to_serialized_data():
            ssh_keys += CustomSshOptions.get_ssh_keys_from_path(
                args.ssh_dest_key_path.to_serialized_data()
            )
```

Reply:
Can you share the code raise this error?
The `AAZList` has `to_serialized_data` function, but the python build-in `list` doesn't have `to_serialized_data` function. The output of `to_serialized_data` for `AAZList` is a python `list`.

Reply:
<img width="556" alt="Error" src="https://github.com/Azure/azure-cli-extensions/assets/11702369/710131dc-038c-49fe-8140-694fd9bf3a62">
This is what we see, please let us know if we are missing anything.